### PR TITLE
WV-2463, WV-2460, WV-2400, WV-2462: Granule layer fixes

### PR DESCRIPTION
**Pull request by jasontk19**
## Description

WV-2463: Filter out granules that do not have associated GIBS imagery based on longitudinal bounds

WV-2460: Fix date parsing when displaying available imagery date ranges

WV-2400: Use a layer's visualization metadata from GIBS to determine what collection to query CMR for

WV-2462: Footprints don't always show on page load

## How To Test

You will need to have the GIBS layer config repository cloned locally to test.  You will also need to have docker running an httpd container with the layer metadata directory mounted.

```
// get the httpd image
docker pull httpd

// start a container named gibs-config which serves the contents of the layers-config repo on localhost:8080
docker run --name gibs-config -d -p 8080:80 -v  /<your_path>/layers-config:/usr/local/apache2/htdocs httpd
```

* Checkout `granule-configs`
* Change the `features.json` `vismetadata` `url` property to `"http://localhost:8080/layer-metadata/v1.0/"`
* `npm run build`
* Checkout this branch
* `npm run watch`

WV-2463:
* Load a granule layer
* Step around through time and confirm that granule footprints only show over imagery within the defined longitudinal bounds (-65 to 65 in geographic)

WV-2460:
* Confim that the start and end date availability as shown in the layer info panel matches what is seen when setting the timeline to those datetimes.  
** (e.g. if the end time is 2022-10-12T00:48Z then set the timeline to this and confirm that imagery still shows but not after this time)

WV-2400:
* If granules are loading, this is working properly

WV-2462:
* Open a link that should have granule imagery visible ([like this](http://localhost:3000/?z=4&ics=true&ici=5&icd=10&l=VIIRS_NOAA20_CorrectedReflectance_TrueColor_Granule,Reference_Labels_15m(hidden),Reference_Features_15m(hidden),Coastlines_15m,VIIRS_NOAA20_CorrectedReflectance_TrueColor(hidden),VIIRS_SNPP_CorrectedReflectance_TrueColor(hidden),MODIS_Terra_CorrectedReflectance_TrueColor(hidden)&lg=true&t=2022-10-10-T12%3A45%3A00Z))
* When page loads, confirm granule footprints are visible on hover
* If possible, try in a new incognito window or other various browsers

### CODE REVIEW

**Review by minniewong — APPROVED** [review]
🚀 looking good!

**Comment by jasontk19** [comment]
> I'm not seeing any imagery prior to Oct 4 for these 3 layers (I cleared out the URL params to test):
> 
> Granule Corrected Reflectance (Bands M11-I2-I1) - (Temporal Coverage: 2022 SEP 26 22:48Z-2022 OCT 11 22:48Z) Granule Corrected Reflectance (Bands M3-I3-M11) - (Temporal Coverage: 2022 SEP 26 05:54Z-2022 OCT 11 05:54Z_ **I only see granules from Oct 4 to Oct 11**
> 
> Granule Corrected Reflectance (True Color) - (Temporal Coverage: 2022 SEP 26 22:48Z-2022 OCT 11 22:48Z) **I only see granules from Oct 5 to Oct 11**

There is a separate ticket to address this.  As it is currently, the dates will only be accurate if you have _just_ done a build.

**Review by ryanweiler92 — APPROVED** [review]

File: web/js/map/granule/util.js [hunk 1]
```diff
@@ -107,23 +129,64 @@ export const getGranuleFootprints = (layer) => {
...
+  }
+  // Polar projections
+  return {
+    startQueryDate: util.dateAdd(date, 'hour', -48),
+    endQueryDate: util.dateAdd(date, 'hour', 4),
+  };
+};
+
+/**
+ * Get the URL parameters for a CMR request for granule browse
```
Comment (Tomcariello) [inline web/js/map/granule/util.js:157]: The params listed in the definition (date & state) are inconsistent with the parameters required for the getParamsForGranuleRequest function (def, date, crs).

**Review by Tomcariello — APPROVED** [review]
Approved!

**Pull request merged**